homeassistant/components/litejet/config_flow.py

```diff
@@ -59,12 +59,9 @@ async def async_step_user(
         if user_input is not None:
             port = user_input[CONF_PORT]
 
-            await self.async_set_unique_id(port)
-            self._abort_if_unique_id_configured()
-
             try:
-                system = pylitejet.LiteJet(port)
-                system.close()
+                system = await pylitejet.open(port)
+                await system.close()
```

Comment:
I think the `close` statement should have been inside `else` statement

Reply:
I think it would work the same either way. If nothing is raised, then it is equivalent to have the close in the try or in the else. If something is raised, then there is no system object to close.

Reply:
I agree that both would work the same way.
However, the policy in Home Assistant is to only have the bits that can raise inside the `try...except`.
It should have been spotted earlier, but I suggest to adjust now that it has been spotted.

Reply:
Indeed I see that buried in the PEP style guide. I'll fix it in this PR since I'm touching the line of code anyway.